sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -223,6 +238,21 @@ object GpuParquetPartitionReaderFactoryBase {
   private val SPARK_VERSION_METADATA_KEY = "org.apache.spark.version"
   // Copied from Spark
   private val SPARK_LEGACY_DATETIME = "org.apache.spark.legacyDateTime"
+  // Copied from Spark
+  private val SPARK_LEGACY_INT96 = "org.apache.spark.legacyINT96"
+
+  def isCorrectedInt96RebaseMode(
+      lookupFileMeta: String => String,
+      isCorrectedModeConfig: Boolean): Boolean = {
+    // If there is no version, we return the mode specified by the config.
+    Option(lookupFileMeta(SPARK_VERSION_METADATA_KEY)).map { version =>
+      // Files written by Spark 3.0 and earlier follow the legacy hybrid calendar and we need to
+      // rebase the INT96 timestamp values.
+      // Files written by Spark 3.1 and latter may also need the rebase if they were written with
+      // the "LEGACY" rebase mode.
+      version >= "3.1.0" && lookupFileMeta(SPARK_LEGACY_INT96) == null
```

Comment:
Doesn't this need to also behave like `isCorrectedRebaseMode` if the recorded Spark version is < "3.1.0"?

Reply:
Yes, and it will because `int96RebaseMode` is based on the `dateTimeRebaseMode` if Spark version < 3.1.0

Reply:
Sorry, maybe I'm missing something here.  It looks like this method is being unconditionally called, with the result stored in a variable `isCorrectedInt96RebaseForThisFile`, to determine the setting for this file.  If this is called on a file written by Spark 3.0 then this will always return false because it will fail the first `version >= "3.1.0"` clause.  For files written by Spark versions >= 3.0 and < 3.1, I think this needs to be:
```
  (version >= "3.1.0" && lookupFileMeta(SPARK_LEGACY_INT96) == null)
    || (version >= "3.0.0" && lookupFileMeta(SPARK_LEGACY_DATETIME == null)
```
or maybe a bit more elaborately for better readability:
```
  if (version >= "3.0.0") {
    // Starting with Spark 3.1, INT96 has a separate metadata key from datetime conversions
    val metaKey = if (version >= "3.1.0") SPARK_LEGACY_INT96 else SPARK_LEGACY_DATETIME
    lookupFileMeta(metaKey) == null
  } else {
    false
  }
```
This would then not always return false for files written by Spark 3.0.0 since it would lookup the old datetime metadata key being used for INT96 on those files.